recipes/cryptopp-pem/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+import textwrap
+
+required_conan_version = ">=1.43.0"
+
+
+class CryptoPPPEMConan(ConanFile):
+    name = "cryptopp-pem"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.cryptopp.com/wiki/PEM_Pack"
+    license = "Unlicense"
+    description = "The PEM Pack is a partial implementation of message encryption which allows you to read and write PEM encoded keys and parameters, including encrypted private keys."
+    topics = ("cryptopp", "crypto", "cryptographic", "security", "PEM")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        suffix = "CRYPTOPP_{}".format(self.version.replace(".", "_"))
+
+        # Get sources
+        tools.get(**self.conan_data["sources"][self.version]["source"],
+                  strip_root=True, destination=self._source_subfolder)
+
+        # Get CMakeLists
+        tools.get(**self.conan_data["sources"][self.version]["cmake"])
+        src_folder = os.path.join(self.source_folder, "cryptopp-cmake-" + suffix)
+        dst_folder = os.path.join(self.source_folder, self._source_subfolder)
+        shutil.move(os.path.join(src_folder, "CMakeLists.txt"), os.path.join(dst_folder, "CMakeLists.txt"))
+        shutil.move(os.path.join(src_folder, "cryptopp-config.cmake"), os.path.join(dst_folder, "cryptopp-config.cmake"))
+        tools.rmdir(src_folder)
+        
+        # Get license
+        tools.download("https://unlicense.org/UNLICENSE", "UNLICENSE")
```

Comment:
```suggestion
        tools.download("https://unlicense.org/UNLICENSE", "UNLICENSE", sha256="7e12e5df4bae12cb21581ba157ced20e1986a0508dd10d0e8a4ab9a4cf94e85c")
```
Usually we add it to conandata.yml (approval-test is a real example), but I think is fine using here, as it's not dynamic.

Reply:
I just discover the github feature of integrating directly the commit suggested in the comment, nice !